src/if-csv/util/helpers.ts

```diff
@@ -0,0 +1,87 @@
+import * as fs from 'fs/promises';
+import {stringify} from 'csv-stringify/sync';
+import {PluginParams} from '@grnsft/if-core/types';
+import {ERRORS} from '@grnsft/if-core/utils';
+
+import {load} from '../../common/lib/load';
+
+import {CsvOptions} from '../types/csv';
+import {STRINGS} from '../config';
+
+const {FAILURE_MESSAGE_OUTPUTS} = STRINGS;
+const {ManifestValidationError} = ERRORS;
+
+/**
+ * Gets the manifest data if `outputs` is present in it.
+ */
+export const getManifestData = async (manifest: string) => {
+  const {rawManifest} = await load(manifest);
+  const {children} = rawManifest.tree;
+
+  if (children?.child?.outputs || children?.['child-0']?.outputs) {
+    return rawManifest;
+  }
+
+  throw new ManifestValidationError(FAILURE_MESSAGE_OUTPUTS);
+};
+
+/**
+ * Executes a CSV generation based on the provided tree structure, context, output path, and params.
+ */
+export const executeCsv = async (options: CsvOptions) => {
```

Comment:
execute csv title doesn't describe what this function does. Can be renamed to extract csv by criteria or something like that